content/client-lib-development-guide/features.textile

```diff
@@ -606,6 +612,13 @@ h4. ClientOptions
 *** @(TO3k3)@ @realtimeHost@ string - for development environments only; allows a non-default Ably Realtime host to be specified
 *** @(TO3k4)@ @port@ integer - for development environments only; allows a non-default Ably non-TLS port to be specified
 *** @(TO3k5)@ @tlsPort@ integer - for development environments only; allows a non-default Ably TLS port to be specified
+** @(TO3l)@ The follow attributes, if set, are used to change the default behaviour of the libary:
+*** @(TO3l1)@ @disconnectedRetryFrequency@ integer - default 15,000 (15s). When the connection is in the @DISCONNECTED@ state, this delay in milliseconds controls how frequently the client library attempts to reconnect automatically
+*** @(TO3l2)@ @suspendedRetryFrequency@ integer - default 30,000 (30s). When the connection is in the @SUSPENDED@ state, this delay in milliseconds controls how frequently the client library attempts to reconnect automatically
+*** @(TO3l3)@ @httpOpenTimeout@ integer - default 4,000 (4s). Timeout for opening the connection, available in the client library if supported by the transport
+*** @(TO3l4)@ @httpRequestTimeout@ integer - default 15,000 (15s). Timeout for any single HTTP request and response
+*** @(TO3l5)@ @httpMaxRetryCount@ integer - default 3. Max number of fallback host retries for HTTP requests that fail due to network issues or server problems
+*** @(TO3l6)@ @httpMaxRetryDuration@ integer - default 10,000 (10s). Max elapsed time in which fallback host retries for HTTP requests will be attempted i.e. if the first default host attempt takes 5s, and then the subsequent fallback retry attempt takes 7s, no further fallback host attempts will be made as the total elapsed time of 12s exceeds the default 10s limit
```

Comment:
Any such requirement would only apply to transports that allow you to implement it. In JSONP you can do even less than XHR.

However ... I think the only place where it could potentially be useful to get an indication of a failure to connect in a shorter time than the expected time for the request itself to complete is a non-streaming comet `recv`. This of course uses just those transports that can't give you the feedback you'd want.